### PR TITLE
FOGL-7787: Control dispatcher components access provided based on user role

### PR DESCRIPTION
**Pull request by hemantgarg923**

### CODE REVIEW

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-write/add-write.component.ts [hunk 1]
```diff
@@ -21,6 +21,7 @@ export class AddWriteComponent implements OnInit {
     private servicesApiService: ServicesApiService,
     private alertService: AlertService,
     public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-write/add-write.component.ts:23]: remove

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-step.component.ts [hunk 2]
```diff
@@ -29,7 +29,8 @@ export class AddStepComponent implements OnInit {
     private alertService: AlertService,
     private controlService: ControlDispatcherService,
     private ngProgress: ProgressBarService,
-    public sharedService: SharedService) {
+    public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-step.component.ts:32]: same as above

File: src/app/components/core/control-dispatcher/add-control-acl/add-control-acl.component.ts [hunk 3]
```diff
@@ -42,7 +42,8 @@ export class AddControlAclComponent implements OnInit {
     private dialogService: DialogService,
     private docService: DocService,
     private servicesApiService: ServicesApiService,
-    public sharedService: SharedService) { }
+    public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-acl/add-control-acl.component.ts:45]: `sharedService` can be removed from here. There is no use of `SharedService` in this file now.

File: src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts [hunk 4]
```diff
@@ -40,6 +40,7 @@ export class AddControlScriptComponent implements OnInit {
     private dialogService: DialogService,
     public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-control-script.component.ts:41]: Can be removed from here as well.

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-configure/add-configure.component.ts [hunk 5]
```diff
@@ -42,6 +42,7 @@ export class AddConfigureComponent implements OnInit {
     private alertService: AlertService,
     private configurationService: ConfigurationService,
     public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-configure/add-configure.component.ts:44]: from here also

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-delay/add-delay.component.ts [hunk 6]
```diff
@@ -15,7 +15,7 @@ export class AddDelayComponent implements OnInit {
   @Input() step; // type of step
   @Input() update = false;
 
-  constructor(public control: NgForm, public sharedService: SharedService) { }
+  constructor(public control: NgForm, public sharedService: SharedService, public rolesService: RolesService) { }
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-delay/add-delay.component.ts:18]: same as above

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-operation/add-operation.component.ts [hunk 7]
```diff
@@ -20,6 +20,7 @@ export class AddOperationComponent implements OnInit {
     private servicesApiService: ServicesApiService,
     private alertService: AlertService,
     public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-operation/add-operation.component.ts:22]: same as above

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-script/add-script.component.ts [hunk 8]
```diff
@@ -27,6 +27,7 @@ export class AddScriptComponent implements OnInit {
     private alertService: AlertService,
     private controlService: ControlDispatcherService,
     public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-script/add-script.component.ts:29]: same as above

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-step-condition/add-step-condition.component.ts [hunk 9]
```diff
@@ -19,7 +19,8 @@ export class AddStepConditionComponent implements OnInit {
   showConditionControl = false;
 
   constructor(private control: NgForm,
-    public sharedService: SharedService) { }
+    public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-step-condition/add-step-condition.component.ts:22]: same as above

File: src/app/components/core/control-dispatcher/add-control-script/add-step/add-step-value/add-step-value.component.ts [hunk 10]
```diff
@@ -17,7 +17,8 @@ export class AddStepValueComponent implements OnInit {
   @Input() update = false;
 
   constructor(private control: NgForm,
-    public sharedService: SharedService) { }
+    public sharedService: SharedService,
```
Comment (monika-Dianomic) [inline src/app/components/core/control-dispatcher/add-control-script/add-step/add-step-value/add-step-value.component.ts:20]: same as above

**Review by monika-Dianomic — APPROVED** [review]

**Comment by mshariq-nerd** [comment]
This is fixed by https://github.com/fledge-iot/fledge-gui/pull/318. Closing

**Pull request closed without merging**